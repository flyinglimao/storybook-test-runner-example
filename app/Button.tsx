import { useState } from "react";

export function Button({ onClick }: { onClick: (cnt: number) => void }) {
  const [count, setCount] = useState(0);
  return (
    <button
      onClick={() => {
        setCount(count + 1);
        onClick(count + 1);
      }}
    >
      {count}
    </button>
  );
}
