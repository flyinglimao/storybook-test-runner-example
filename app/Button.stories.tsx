import type { Meta, StoryObj } from "@storybook/react";
import { expect, fn, userEvent, within } from "@storybook/test";

import { Button } from "./Button";

const meta: Meta<typeof Button> = {
  title: "Button",
  component: Button,
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    onClick: fn(),
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    await window.takeSnapshot?.();
    await window.takeScreenshot?.();

    await userEvent.click(canvas.getByRole("button"));
    await expect(args.onClick).toHaveBeenLastCalledWith(1);

    await window.takeSnapshot?.();
    await window.takeScreenshot?.();
  },
};
