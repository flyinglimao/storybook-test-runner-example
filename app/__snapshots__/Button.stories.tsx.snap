// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Primary play-test 1`] = `
<button>
  0
</button>
`;

exports[`Button Primary play-test 3`] = `
<button>
  1
</button>
`;
